flambe/experiment/utils.py

```diff
@@ -549,3 +550,47 @@ def shutdown_remote_ray_node(host: str,
     """
     cmd = f"ssh -i {key} -o StrictHostKeyChecking=no {user}@{host} \"bash -lc 'ray stop'\""
     return os.system(cmd)
+
+
+def get_default_devices(debug: bool = False,
+                        default_cpus: int = 1,
+                        default_gpus: int = 1) -> Dict[str, int]:
+    """Get the default devices to use if not provided.
+
+    Parameters
+    ----------
+    debug: bool, optional
+        If we are running in debug mode (where Ray is not available).
+        If debug is False, this method should be called after running
+        ``ray.init()``.
+    default_cpus: int, optional
+        The default number of CPU's to use. Default ``1``.
+    default_gpus: int, optional
+        The default number of GPU's to use. Default ``1``.
+
+    Returns
+    -------
+    devices: Dict[str, int]
+        Default set of devices to use. Should have at most two keys:
+        'cpu', and 'gpu' if cuda is available.
+
+    """
+    if debug:
+        if torch.cuda.is_available():
+            devices = {"cpu": default_cpus, "gpu": default_gpus}
+        else:
+            devices = {"cpu": default_cpus}
+    else:
+        if ray.is_initialized():
+            cluster_devices = ray.cluster_resources()
+            use_gpu = 'GPU' in cluster_devices or 'gpu' in cluster_devices
+        elif torch.cuda.is_available():
```

Comment:
I would just fail if ray wan't initialized

Reply:
to not make the logic more complex

Reply:
disagree, this function is more powerful as is!